src/main/java/org/reactome/addlinks/fileprocessors/RGDFileProcessor.java

```diff
@@ -0,0 +1,64 @@
+package org.reactome.addlinks.fileprocessors;
+
+import org.reactome.addlinks.EnsemblBiomartUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RGDFileProcessor extends FileProcessor{
+
+    public RGDFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public RGDFileProcessor()
+    {
+        super();
+    }
+
+    private static final int rgdIdentifierIndex = 0;
+    private static final int uniprotIdentifiersIndex = 21;
+
+    /**
+     * Build map of UniProt identifiers to Rat Genome Database (RGD) identifiers that is used to create RGD cross-references in database.
+     * @return - Map<String, List<String>
+     */
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFilePath = Paths.get(this.pathToFile.toAbsolutePath().toString());
+
+        List<String> lines = new ArrayList<>();
+        try {
+            lines = EnsemblBiomartUtil.getLinesFromFile(inputFilePath, true);
+        } catch (IOException e) {
+            logger.error("Error reading file ({}): {}", inputFilePath.toString(), e.getMessage());
+            e.printStackTrace();
+        }
+
+        for (String line : lines) {
+            if (isFileBodyLine(line)) {
+                List<String> tabSplit = Arrays.asList(line.split("\t"));
+                String rgdId = tabSplit.get(rgdIdentifierIndex);
+                List<String> uniprotIds = Arrays.asList(tabSplit.get(uniprotIdentifiersIndex).split(";"));
+                if (!uniprotIds.isEmpty()) {
```

Comment:
This check isn't needed, I don't think.  `Arrays.asList()` should always return an ArrayList object and the for loop inside the if block will not be executed if the list is empty.